stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -309,37 +309,49 @@ internal class StripePaymentController internal constructor(
         }
     }
 
-    private fun onSourceRetrieved(
+    private suspend fun onSourceRetrieved(
         host: AuthActivityStarter.Host,
         source: Source,
         requestOptions: ApiRequest.Options
     ) {
         if (source.flow == Source.Flow.Redirect) {
-            analyticsRequestExecutor.executeAsync(
-                analyticsRequestFactory.create(
-                    analyticsDataFactory.createAuthSourceParams(
-                        AnalyticsEvent.AuthSourceRedirect,
-                        source.id
-                    )
-                )
-            )
-
-            paymentAuthWebViewStarterFactory(host).start(
-                PaymentAuthWebViewContract.Args(
-                    objectId = source.id.orEmpty(),
-                    requestCode = SOURCE_REQUEST_CODE,
-                    clientSecret = source.clientSecret.orEmpty(),
-                    url = source.redirect?.url.orEmpty(),
-                    returnUrl = source.redirect?.returnUrl,
-                    enableLogging = enableLogging,
-                    stripeAccountId = requestOptions.stripeAccount
-                )
+            startSourceAuth(
+                paymentAuthWebViewStarterFactory(host),
+                source,
+                requestOptions
             )
         } else {
             bypassAuth(host, source, requestOptions.stripeAccount)
         }
     }
 
+    private suspend fun startSourceAuth(
+        paymentAuthWebViewStarter: PaymentAuthWebViewStarter,
+        source: Source,
+        requestOptions: ApiRequest.Options
+    ) = withContext(Dispatchers.Main) {
```

Comment:
IMO we should always pass the coroutine context/dispatcher to the constructor. In this case the `Main` dispatcher is always replaced in unit tests with the test dispatcher but that is not the case for other dispatchers. Not something that should be changed now, just something to keep in mind.